backend/src/core/services/credential.service.ts

```diff
@@ -62,12 +68,17 @@ const upsertCredential = async (
  * Save a credential into secrets manager and the credential table
  * @param name
  * @param secret
+ * @param isBotToken
  */
-const storeCredential = async (name: string, secret: string): Promise<void> => {
+const storeCredential = async (
+  name: string,
+  secret: string,
+  isBotToken = false
```

Comment:
Minor nitpick. Was thinking if we can rename this argument to something that is more generic and expresses the intent better. Something along the lines of `restrictEnvironment` (?). Otherwise someone without familiarity with this PR might be wondering why bot tokens are treated differently.

Reply:
Good idea, changed it!